packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -1,16 +1,206 @@
-## AWS::Synthetics Construct Library
+## Amazon CloudWatch Synthetics Construct Library
+
 <!--BEGIN STABILITY BANNER-->
 ---
 
 ![cfn-resources: Stable](https://img.shields.io/badge/cfn--resources-stable-success.svg?style=for-the-badge)
 
 > All classes with the `Cfn` prefix in this module ([CFN Resources](https://docs.aws.amazon.com/cdk/latest/guide/constructs.html#constructs_lib)) are always stable and safe to use.
 
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+> The APIs of higher level constructs in this module are experimental and under active development. They are subject to non-backward compatible changes or removal in any future version. These are not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be announced in the release notes. This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+Amazon CloudWatch Synthetics allow you to monitor your application by generating **synthetic** traffic. The traffic is produced by a **canary**. A canary is a configurable script that runs on a schedule. You configure the canary script to follow the same routes and perform the same actions as a customer, which allows you to continually verify your customer experience even when you don't have any customer traffic on your applications.
+
+## Canary
+
+To illustrate how to use a canary, assume your application defines the following endpoint:
+
+```bash
+curl https://api.awsomesite.com/user/books/topbook/
+The Hitchhikers Guide to the Galaxy
+
+```
+
+The below code defines a canary that will hit the `books/topbook` endpoint every 5 minutes:
+
+> **Not Implemented:** the `test` property will be a part of Milestone 2.
+
+```ts
+const canary = new Canary(this, 'MyCanary', {
+  name: 'mycanary'
+  frequency: Duration.minutes(5),
+  test: synth.Test.custom({
+    code: synth.Code.fromInline(`const https = require('https');
```

Comment:
Is the `synth.Code` the same than `@aws-cdk/aws-lambda.Code`? If they differ - is there a way they could be unified?

Reply:
You mean moving `Code` to core?
The biggest difference is that in the synthetics module, if the code is stored in s3,  the synthetics `Code` property requires that the zip file will have a specific structure, from their [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_WritingCanary.html):

>If you have more than a single .js file or you have a dependency that your script depends on, you can bundle them all into a single ZIP file that contains the folder structure nodejs/node_modules/myCanaryFilename.js file and other folders and files.

(Note that the "If you have more than" is misleading as this is always required)

If the zip file does not conform to the describe structure the deployment will succeed but **the canary execution will fail** , possibly triggering an alarm. To spare this from users the synthetics `Code` class should include the logic that zip the local directory (asset).